numba_dpex/tests/dpjit_tests/dpnp/test_dpnp_empty.py

```diff
@@ -38,17 +39,40 @@ def func(shape):
 
     assert c.dtype == dtype
     assert c.usm_type == usm_type
-    if device != "unknown":
-        assert (
-            c.sycl_device.filter_string
-            == dpctl.SyclDevice(device).filter_string
-        )
+    assert c.sycl_device.filter_string == device
+
+
+@pytest.mark.parametrize("shape", shapes)
+@pytest.mark.parametrize("dtype", dtypes)
+@pytest.mark.parametrize("usm_type", usm_types)
+def test_dpnp_empty_from_queue(shape, dtype, usm_type):
+    @dpjit
+    def func(shape, queue):
+        c = dpnp.empty(shape, dtype=dtype, usm_type=usm_type, sycl_queue=queue)
+        return c
+
+    queue = dpctl.SyclQueue()
+
+    try:
+        c = func(shape, queue)
+    except Exception:
+        pytest.fail("Calling dpnp.empty inside dpjit failed")
+
+    if len(c.shape) == 1:
+        assert c.shape[0] == shape
     else:
-        c.sycl_device.filter_string == dpctl.SyclDevice().filter_string
+        assert c.shape == shape
+
+    assert c.dtype == dtype
+    assert c.usm_type == usm_type
+    assert c.sycl_device.filter_string == queue.sycl_device.filter_string
```

Comment:
Why are we comparing on `filter_string`? Why don't we do just `assert c.sycl_device == queue.sycl_device`?

Reply:
We can try. The reason is that they device objects will not be the same. It depends on how the `dpctlSyclDevice.__eq__` is implemented.